tutorials/embedded/step-7.html.haml

```diff
@@ -0,0 +1,208 @@
+- @tutorial_steps = [ "The Weather Application", "Initializing the CacheManager", "Putting stuff in the cache", "Making entries expire", "Configuring the default cache", "Clustering the application", "Listen to changes in the cluster", "Listen to changes in the cache", "Grouping entries together", "A custom externalizer", "Temperature averages with map/reduce", "Declarative configuration" ]
+- tutorial_step = 7
+.row-fluid
+  %h1 Infinispan Embedded Tutorial: The Weather App
+  #columns
+    #leftcol.span3.well
+      = partial( 'tutorialmenu.html.haml', { "tutorial_steps" => @tutorial_steps, "tutorial_step" => tutorial_step } )
+
+    #maincol.span9
+      = partial( 'tutorialnav.html.haml', { "tutorial_steps" => @tutorial_steps, "tutorial_step" => tutorial_step, "tutorial_github_url" => tutorial_github_url, "header" => true } )
+      %p
+        Now that we know how to react on changes in the cluster topology, we can also react to changes to the data within the cluster. To better show how this work, we will separate
+        the roles of our two nodes: the coordinator will take care of putting data in the cache, and the other node will show the cache modifications. 
+        %pre
+          %code.java
+            = preserve do
+              :escaped
+                @Listener(clustered = true)
+                public class CacheListener {
+                   @CacheEntryCreated
+                   public void entryCreated(CacheEntryCreatedEvent<String, LocationWeather> event) {
+                      if (!event.isOriginLocal()) {
+                         System.out.printf("-- Entry for %s modified by another node in the cluster\n", event.getKey());
+                      }
+                   }
+                }
+        Usually a cache listener only listens to "local events", i.e. events which happen on the same node. For our example, however, we want to listen to events which happen
+        on all nodes, and therefore we use the
+        %i "clustered = true"
+        parameter on the annotation. Let's see the event notifications:
+        %pre
+          %code.bash
+            = preserve do
+              :escaped
+                $ git checkout -f step-7
+                $ mvn clean package exec:exec # on terminal 1
+                $ mvn exec:exec # on terminal 2
+		
+        #output.accordion
+          .accordion-group
+            .accordion-heading
+              %a.accordion-toggle{"data-parent" => "#output", "data-toggle" => "collapse", :href => "#collapseOne"}
+                Coordinator Node Output
+                %i.icon-angle-down
+              #collapseOne.accordion-body.collapse
+                .accordion-inner
+                  %pre
+                    %code.bash
+                      = preserve do
+                        :escaped        
+                          Jan 30, 2015 3:22:08 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport start
```

Comment:
I think it would be good to drop the ISPN/JGroups log entries completely to make the output more readable.